### ISSUE TITLE
Syntax Error on Ubuntu v. 16 (ChakraCore v. 1.3.1)

### ISSUE THREAD

**Issue by SBuder** [Microsoft/ChakraCore#2141]
When executing JS script on Linux we get an error saying "SyntaxError: Invalid character"

Script|Succeeds|comment
---|---|---
$global.data[1];|false|Seems to fail due to semicolon|
$global.data[1]|true|Succeeds without semicolon|
function foo(){ var a = [true,false]; return a; } |true|
function foo(){ var a = [1,2]; return a; }|true|This works, but when it comes to an objects array it again raises an error
function foo(){ var a = [{'b': true}, {'b': false}]; return a; }|false|Again with missing semicolon its working
function foo(){ var a = [{'b': true}, {'b': false}]; return a }|true|this only happens when using an object array 

Basically we are using native method: JsRunScriptUtf8 to execute scripts above. Can you please give me a hint what we are doing wrong, since its seems to be a pretty weird. All scripts mentioned do not raise any error when running them on windows machines.

In addtion to that can you please provide in more detailed example on how to get to the exact error from the engine. I already made it to get the error message but the line and column number would be great to include into the error.

Thanks in advance.

**Comment by obastemur** [comment]
@SBuder I suspect a problem with measuring the string length. If you could share some part of your code, that could help us to identify.

**Comment by SBuder** [comment]
```cs
[DllImport( "libChakraCore.so" )]
internal static extern JavaScriptErrorCode JsRunScriptUtf8(
      byte[] script,
      JavaScriptSourceContext sourceContext,
      byte[] sourceUrl,
      out JavaScriptValue result );
```
We use this method of the native library, by calling it like this:

```cs
internal static JavaScriptErrorCode JsRunScript( string script,
      JavaScriptSourceContext sourceContext,
      string sourceUrl,
      out JavaScriptValue result ) {
      var scriptUtf8Bytes = Encoding.UTF8.GetBytes( script );
      var sourceUrlUtf8Bytes = Encoding.UTF8.GetBytes( sourceUrl );
      
return JsRunScriptUtf8(scriptUtf8Bytes, sourceContext, sourceUrlUtf8Bytes, out result );
}
```

**Comment by SBuder** [comment]
I was able to solve this issue by terminating the string / UTF8 bytes explicitly with a 0

So something like:

```cs
 var scriptUtf8Bytes = Encoding.UTF8.GetBytes( script ).ToList();
scriptUtf8Bytes.Add( 0 );

var sourceUrlUtf8Bytes = Encoding.UTF8.GetBytes( sourceUrl ).ToList();
sourceUrlUtf8Bytes.Add( 0 );
```

Would work.
